hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/AllocationConfiguration.java

```diff
@@ -293,26 +293,15 @@ Resource getMinResources(String queue) {
 
   /**
    * Get the maximum resource allocation for the given queue. If the max in not
-   * set, return the larger of the min and the default max.
+   * set, return the default max.
```

Comment:
I changed it since this is not the right place to do that. I move this behavior to FSQueue#getMaxShare. The reason is allocation file reload and cluster resource changing are two independent events. It is not right to calculate based on cluster resource while refresh the allocation file.